safekeeper/src/http/routes.rs

```diff
@@ -126,7 +126,7 @@ async fn timeline_create_handler(mut request: Request<Body>) -> Result<Response<
     let request_data: TimelineCreateRequest = json_request(&mut request).await?;
 
     let zttid = ZTenantTimelineId {
-        tenant_id: request_data.tenant_id,
+        tenant_id: parse_request_param(&request, "tenant_id")?,
```

Comment:
Having `tenant_id` in the url and `timeline_id` in the body looks strange. I'm interested what's the reasoning behind this? It seems that in all other places they are together in the url like `/tenant/:tenant_id/timeline/:timeline_id`.

Reply:
Maybe I'm wrong, but I think that it's just usual REST convention:
- when you use `/resources/:id` you address some existing resource, so it's usually GET, PATCH, DELETE
- when you create new resource, you use `POST or PUT /resources/`, because i) you may put an arbitrary payload as body and even if `id` is known at creation time it doesn't have much sense to split a single entity payload into path and body parts; and ii) `id` is usually not known at creation time (not our case, though)

Reply:
BTW, as far as I see, this endpoint is never used. And anyway we don't have tenant creation endpoint, so does tenant represent some separate entity on pageserver side? Or it always goes along with timeline? So it doesn't matter whether there are some timelines for this tenant already or not